pyperf/_runner.py

```diff
@@ -501,6 +522,8 @@ def bench_func(self, name, func, *args, **kwargs):
         if not self._check_worker_task():
             return None
 
+        func = stats_wrapper(func)
```

Comment:
On a Python built with sys._stats_on(), this wrapper adds a fixed overhead. If you want to execute code before/after the benchmark, it should be done before/after code measuring time.

See task_func() below.

Reply:
In this case, the benchmarking timings are useless anyway (due to the stats overhead), but it would have a (minor) impact on the statistics themselves.  I'll see if I can refactor this, though.  It probably requires a copy-and-paste of `task_func` to really minimize the overhead -- I'm not sure it's worth it, to be honest.

Reply:
If we are gathering stats, the timings are meaningless.
To maximize the accuracy of the stats, we want to turn them on and off as close to the function call as possible.
This looks about as good as we can get.